libs/api/ebpf_api.cpp

```diff
@@ -3729,7 +3729,10 @@ ebpf_ring_buffer_map_subscribe(
         if (result == EBPF_PENDING)
             result = EBPF_SUCCESS;
 
-        *subscription = local_subscription.release();
+        // If the async IOCTL failed, then free the subscription object.
```

Comment:
This comment is a bit confusing.  The immediately following ```if``` block 'releases' the subscription on success, so I'm assuming 'release' does not mean 'free'.  In that case, I don't see the subscription block being 'freed' on failure either?

Reply:
Release in this context is: [std::unique_ptr::release](https://cplusplus.com/reference/memory/unique_ptr/release/) which releases ownership of the underlying memory and assigns it to the caller.

Reply:
resolving per out follow-up conversation.